searx/engines/google_news.py

```diff
@@ -18,72 +52,155 @@
+    logger.debug("query_url --> %s", query_url)
+
+    # en-US,en;q=0.8,en;q=0.5
+    params['headers']['Accept-Language'] = (
+        lang_country + ',' + language + ';q=0.8,' + language + ';q=0.5'
+        )
+    logger.debug("HTTP header Accept-Language --> %s",
+                 params['headers']['Accept-Language'])
+    params['headers']['Accept'] = (
+        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
+        )
+    #params['google_subdomain'] = subdomain
 
     return params
 
 
-# get response from search-request
 def response(resp):
+    """Get response from google's search request"""
     results = []
 
+    # detect google sorry
+    resp_url = urlparse(resp.url)
+    if resp_url.netloc == 'sorry.google.com' or resp_url.path == '/sorry/IndexRedirect':
+        raise SearxEngineCaptchaException('sorry.google.com')
+
+    if resp_url.path.startswith('/sorry'):
+        raise SearxEngineCaptchaException(gettext('CAPTCHA required'))
+
+    # which subdomain ?
+    # subdomain = resp.search_params.get('google_subdomain')
+
+    # convert the text to dom
     dom = html.fromstring(resp.text)
 
-    # parse results
-    for result in dom.xpath('//div[@class="g"]|//div[@class="g _cy"]'):
-        try:
-            r = {
-                'url': result.xpath('.//a[@class="l lLrAF"]')[0].attrib.get("href"),
-                'title': ''.join(result.xpath('.//a[@class="l lLrAF"]//text()')),
-                'content': ''.join(result.xpath('.//div[@class="st"]//text()')),
-            }
-        except:
-            continue
-
-        imgs = result.xpath('.//img/@src')
-        if len(imgs) and not imgs[0].startswith('data'):
-            r['img_src'] = imgs[0]
-
-        results.append(r)
+    for result in eval_xpath_list(dom, '//div[@class="xrnccd"]'):
+
+        # The first <a> tag in the <article> contains the link to the
+        # article The href attribute of the <a> is a google internal link,
+        # we can't use.  The real link is hidden in the jslog attribute:
+        #
+        #   <a ...
+        #      jslog="95014; 4:https://www.cnn.com/.../index.html; track:click"
+        #      href="./articles/CAIiENu3nGS...?hl=en-US&amp;gl=US&amp;ceid=US%3Aen"
+        #      ... />
+
+        jslog = eval_xpath_getindex(result, './article/a/@jslog', 0)
+        url = re.findall('http[^;]*', jslog)
+        if url:
+            url = url[0]
+        else:
+            # The real URL is base64 encoded in the json attribute:
+            # jslog="95014; 5:W251bGwsbnVsbCxudW...giXQ==; track:click"
+            jslog = jslog.split(";")[1].split(':')[1].strip()
+            try:
+                jslog = b64decode(jslog)
+            except binascii.Error:
+                # Sometimes (very rare) the base64 encoded string is not a
+                # multiple of four.
+                try:
+                    # fix incorrect padding (if possible)
+                    padding = (4 -(len(jslog) % 4)) * "="
```

Comment:
even if the padding is most of time empty, it could be the general case:
```python
try:
  padding = (4 -(len(jslog) % 4)) * "="
  jslog = b64decode(jslog + padding)
except binascii.Error:
  continue
```
?

So the code is simpler

Reply:
Worth to test .. I will try ...